keel-core/src/test/kotlin/com/netflix/spinnaker/keel/services/ComparableLinksTests.kt

```diff
@@ -339,82 +319,49 @@ class ComparableLinksTests : JUnit5Minutests {
-          } answers {
-            PublishedArtifact(
-              name = arg<DeliveryArtifact>(2).name,
-              type = arg<DeliveryArtifact>(2).type,
-              version = version1,
-              gitMetadata = GitMetadata(commit = arg<String>(1),
-                commitInfo = Commit(sha = "pinnedVersion", link = "stash"),
-                repo = Repo(name = "keel"),
-                project = "spkr"
-              )
-            )
-          }
+            repository.getVersionInfoInEnvironment(any(), "test", any())
+          } returns listOf(
+            StatusInfoForArtifactInEnvironment(version0, CURRENT, null, clock.instant().minus(Duration.ofHours(2))),
+          )
 
           every {
-            repository.getArtifactVersion(releaseArtifact, version3, ArtifactStatus.RELEASE)
-          } answers {
-            PublishedArtifact(
-              name = arg<DeliveryArtifact>(0).name,
-              type = arg<DeliveryArtifact>(0).type,
-              version = arg<String>(1),
-              gitMetadata = GitMetadata(
-                commit = "pinnedVersion",
-                commitInfo = Commit(sha = "pinnedVersion", link = "stash"),
-                repo = Repo(name = "keel"),
-                project = "spkr"
-              ),
-              buildMetadata = buildMetadata
-            )
-          }
-          every {
-            repository.getArtifactVersion(releaseArtifact, version0, ArtifactStatus.RELEASE)
-          } answers {
-            PublishedArtifact(
-              name = arg<DeliveryArtifact>(0).name,
-              type = arg<DeliveryArtifact>(0).type,
-              version = arg<String>(1),
-              gitMetadata = GitMetadata(
-                commit = "pinnedVersion",
-                commitInfo = Commit(sha = "pinnedVersion", link = "stash"),
-                repo = Repo(name = "keel"),
-                project = "spkr"
-              ),
-              buildMetadata = buildMetadata
-            )
-          }
+            repository.getVersionInfoInEnvironment(any(), "staging", any())
+          } returns listOf(
+            StatusInfoForArtifactInEnvironment(version0, PREVIOUS, version1, clock.instant().minus(Duration.ofHours(4))),
+            StatusInfoForArtifactInEnvironment(version1, CURRENT, null, clock.instant()),
+          )
         }
+
         test ("get the correct compare link when pinning forward (current)") {
+          // pinning to version 3 here
           val summaries = applicationService.getArtifactSummariesFor(application1)
           expectThat(summaries.first())
             .withVersionInEnvironment(version1, "staging") {
-              compareLink.isEqualTo("https://stash/projects/spkr/repos/keel/compare/commits?targetBranch=fnord-1.0.1-h1.b1b1b1b&sourceBranch=pinnedVersion")
+              compareLink.isEqualTo("https://stash/projects/spkr/repos/keel/compare/commits?targetBranch=${version1}&sourceBranch=${version3}")
```

Comment:
nit: The compiler will give you warnings for all these unnecessary curly braces. I think that's annoying, personally (I don't mind the curly braces, makes things easier to read!), but thought I'd point it out.